src/main/java/org/testng/annotations/ITestAnnotation.java

```diff
@@ -66,10 +66,19 @@
 
   void setDataProviderClass(Class<?> v);
 
+  /**
+   * @deprecated - This method stands deprecated as of TestNG 7.0.0.
+   * Please use {{@link #getRetryAnalyzerClass()}} instead.
+   */
+  @Deprecated
   IRetryAnalyzer getRetryAnalyzer();
 
   void setRetryAnalyzer(Class<? extends IRetryAnalyzer> c);
 
+  default Class<? extends IRetryAnalyzer> getRetryAnalyzerClass() {
+    return null;
```

Comment:
Here you must return `getRetryAnalyzer().getClass()`

Reply:
fixed